### ISSUE TITLE
Is `christmastideIncludesTheSeasonOfEpiphany` option relevant?

### ISSUE THREAD

**Issue by emagnier** [pejulian/romcal#212]
Because there is no season of Epiphany. The period of epiphany is part of the Christmas season.

Christmas season is composed of:
1. The octave of Christmas (including the solemnity of Christmas)
2. The days before epiphany
3. The solemnity of epiphany
4. The days after epiphany (depending on the year)
5. The baptism of the Lord

**Comment by tukusejssirs** [comment]
I have somewhat same thoughts about this, as the Christimas Season in `cal1969` should always last until the day before the _Baptism of the Lord_ (this is connected with `christmastideEnds` romcal option, which I think is irrelevant in `cal1969` from the _liturgical_ point of view, although _some_ of the decorations might be left until the _Presentation of the Lord_).

Another (sub)issue is connected with your 2nd point: neither in Latin, nor in Slovak I have seen _days before Epiphany_. AFAIK, they are always described as _days in Christmas Season_ (or similarly).

**Comment by emagnier** [comment]
> I have somewhat same thoughts about this, as the Christimas Season in cal1969 should always last until the day before the Baptism of the Lord (this is connected with christmastideEnds romcal option, which I think is irrelevant in cal1969 from the liturgical point of view, although some of the decorations might be left until the Presentation of the Lord).

I think we can find the answer in the notes from the missal (Universal Norms on the Liturgical Year and the General Roman Calendar). Because the Sunday after Epiphany is the Baptism of the Lord, and the note specifies that:

> 33. Christmas Time runs from First Vespers (Evening Prayer I) of the Nativity of the Lord up to and including the Sunday after Epiphany or after 6 January.

For the cases where there is no Sunday after Epiphany, Baptism of the Lord is celebrated the Monday just after Epiphany, which is covered by the "or after 6 January". This 2nd case is less precisely described, but we already had the answer in the first case (Baptism of the Lord on Sunday).

Moreover there is no first Sunday of ordinary time, since the ordinary time starts the Monday that follows the Sunday after Epiphany, or start the Tuesday if there is no Sunday of Epiphany (when the Baptism of the Lord is celebrated the Monday just after Epiphany, instead of the Sunday after Epiphany).
First celebrated Sunday during the ordinary time is in fact the "2nd Sunday of Ordinary Time".

> 44. Ordinary Time begins on the Monday which follows the Sunday occurring after 6 January and extends up to and including the Tuesday before the beginning of Lent; 

This also means that the `christmastideEnds` option should always be set up to `o` to be compliant with the `cal1969`. The "traditional" value `t` seems to partially follow the instructions and norms of the previous liturgical calendar before 1969 (with a dedicated season of Epiphany), but this is actually out of the scope of romcal which aims to conforms to the revised liturgical calendar approved by Paul VI in Mysterii Paschalis dated 14 February 1969.

This also means that `christmastideEnds` might not be relevant either.
Removing these 2 options will also help to simplify the codebase and make it more robust and tidy.

---

> Another (sub)issue is connected with your 2nd point: neither in Latin, nor in Slovak I have seen days before Epiphany. AFAIK, they are always described as days in Christmas Season (or similarly).

Yes, all day from the Christmas octave to the baptism of the Lord are days of Christmas Season.
But in the missal and the lectionary they are named more precisely as Christmas weekdays before Epiphany, and directly with their date (Die 2 ianuarii...), even in the latin missal. Probably a convenient way to differentiate them from the days in the Christmas octave, and the days after Epiphany. But most importantly because readings and prayers are different during these days ("lectiones ante Epiphaniam Dicendæ").

**Comment by tukusejssirs** [comment]
I agree with everything you have written.

> Yes, all day from the Christmas octave to the baptism of the Lord are days of Christmas Season.
> But in the missal and the lectionary they are named more precisely as Christmas weekdays before Epiphany, and directly with their date (Die 2 ianuarii...), even in the latin missal. Probably a convenient way to differentiate them from the days in the Christmas octave, and the days after Epiphany. But most importantly because readings and prayers are different during these days ("lectiones ante Epiphaniam Dicendæ").

Well, yes, that might be true, but that makes it a bit more complicated. Nevertheless, if the books use two (or even three) ways to called the same days, it might be a good idea to add a romcal option in order to let the user choose the name, that is:

1. name the days as dates (`2 January`);
2. use `before Epiphany` (`Tuesday before Epiphany`);
3. use `in Christmas Season` (`Tuesday in Christmas Season`).

All three variants should be localised in all supported languages.

What do you think, @emagnier?

**Comment by emagnier** [comment]
After a closer look in the missal, I've found that:

- Prayers are spread over each ferial day of the Christmas time (monday, tuesday...), and they are the same both before and after Epiphany.
- Readings are defined per date before Epiphany (2 January, 3 January...), and per ferial day after Epiphany (monday, tuesday...)
- The "before" or "after epiphany" is mostly a convenient way to classify in the missal & the lectionary the readings/prayers before or after Epiphany. But it's not essential to specify it in the title (and the key) of the celebration, especially since it is not used in all translations.

I'm not very enthusiast to give a user choice of conditional key/title name. However as a median solution, I suggest to name:
- `Tuesday, January 2 in Christmas Season` for all ferial days before Epiphany. So we have the day (speficic prayers) and the date (specific readings) of the celebration.
- `Tuesday in Christmas Season` for all ferial days after Epiphany.

With this naming we are sure we don't have twice the same celebration key/title during the same season (naming are different during the Christmas octave), and we still have enough information to attach the right metadata to the right celebration. 

For a convenient usage, the "days before epiphany" and the "days after epiphany" could be specified in a period property (different from the season property), like for "early/later ordinary time". It's free to be use or not, but is still useful for specific needs (like adding a monastic liturgical layer on top of romcal).

**Comment by tukusejssirs** [comment]
> I'm not very enthusiast to give a user choice of conditional key/title name.

Well, I see you try to do your best to make romcal as minimalistic as possible in this area. I just hope you won’t disagree with me on a partially connected issue that the users could set romcal to use full canonisation level (e.g. `Saint`) or just its abbreviation (e.g. `St`), full title names (e.g. `Pope`, `Popes`, `Doctors of the Church`, `Martyr`, `Martyrs`) or their abbreviations (e.g `P`, `Pp`, `D`, `M`, `Mm`). However, that is a different issue I’d like to raise, but I am currently busy with the l10n and celebration definition proofing.

> However as a median solution, I suggest to name:
> 
>     * `Tuesday, January 2 in Christmas Season` for all ferial days before Epiphany. So we have the day (speficic prayers) and the date (specific readings) of the celebration.
> 
>     * `Tuesday in Christmas Season` for all ferial days after Epiphany.

Well, `Tuesday, January 2 in Christmas Season` looks quite strange to me. I have never seen this format, because the date is usually either in the separate column (when the celebrations are listed in a table) or above (when the celebration name is used as a title). I’d rather use the second option only (`Tuesday in Christmas Season`).

> With this naming we are sure we don't have twice the same celebration key/title during the same season (naming are different during the Christmas octave), and we still have enough information to attach the right metadata to the right celebration.

From my point of view, we can still use the date from the JSON that romcal outputs to validate the correctness of the celebration, no need to add the date into the celebration name.

> For a convenient usage, the "days before epiphany" and the "days after epiphany" could be specified in a period property (different from the season property), like for "early/later ordinary time". It's free to be use or not, but is still useful for specific needs (like adding a monastic liturgical layer on top of romcal).

I have no problem with _additional_ properties (keys in the JSON romcal outputs) that would provide additional metadata (even those not actually part of the Roman Liturgy, i.e. those that help the implementation of it), as long as the ‘main’ properties provide the correct (or as correct as possible) information/metadata. Same applies to the celebration names.

**Comment by emagnier** [comment]
> Well, I see you try to do your best to make romcal as minimalistic as possible in this area.

In fact, I want the key to be unique and avoid to have a naming choice. And the key should always try to define the celebration precisely. But then about the title of the celebration (not the key), you are right. Currently the celebration titles are very closed to their keys, but in the future we could allow variations with the help of a localization tool (like "St" vs. "Saint"...).

> Well, `Tuesday, January 2 in Christmas Season` looks quite strange to me.

Maybe the naming could be improved, but during this period (days before epiphany), date **and** day are very important, to reflect the right celebration. And it's also very important to avoid 2 celebrations with the same name and key.
If you have a look to the missal and the lectionary, you will see that these weekdays before Epiphany are defined by their dates.

By the way we have quite a similar situation (I don't had time to open issue yet for that) during the time of Advent: from December 17 to December 24, these weekdays are different of the rest of the weekdays cycle (they overwrite them), they have specific readings and prayers for each date, and they even take precedence over any memorial (memorials are downgraded to commemorations, like in lent). There too it's important to have a key and celebration title that reflect the celebration.

**Comment by tukusejssirs** [comment]
> > Well, I see you try to do your best to make romcal as minimalistic as possible in this area.
> 
> In fact, I want the key to be unique and avoid to have a naming choice. And the key should always try to define the celebration precisely. But then about the title of the celebration (not the key), you are right. Currently the celebration titles are very closed to their keys, but in the future we could allow variations with the help of a localization tool (like "St" vs. "Saint"...).

Okay, do as you see fit. I am impatiently looking forward to the localisation tool! :smiley:

> By the way we have quite a similar situation (I don't had time to open issue yet for that) during the time of Advent: from December 17 to December 24, these weekdays are different of the rest of the weekdays cycle (they overwrite them), they have specific readings and prayers for each date, and they even take precedence over any memorial (memorials are downgraded to commemorations, like in lent). There too it's important to have a key and celebration title that reflect the celebration.

Yes, I have discovered the 17-24 Dec issue while localising the keys into Latin (I have a note to create an issue for it, but I have about 250-300 notes on what needs some change, incl some issues I can’t fix myself). If we want to have the same celebration name format for the days before (and probably after) Epiphany as we should have on 17-24 Dec, we could use simply `2 January` as the title for those ferias.

**Comment by tukusejssirs** [comment]
@emagnier, I have found that in the [2018 Ordo of New Zealand](https://www.nlo.org.nz/assets/Uploads/Aotearoa-NZ-Ordo-January-February-2018.pdf), they used `Weekday of Christmas Time` on 3 January.

However, if you still wish to include the date, I suggest to format the celebration name as follows: `3 January, Wednesday of Christmas Time` (or you might replace the comma with a colon). Of course, the date (`3 January`) has to be localised (e.g., in the US, they would prefer `January 3`).

**Update:** Replaced `Weekday` to `Wednesday`.

**Comment by emagnier** [comment]
Liturgical ordo are always interesting, but they are made only in a context of a specific year, for a specific region/diocese only, and could eventually contain some adaptations or mistakes by the guys doing this job every years in every dioceses (even if a big proofreading work is done every time). Ordo are a great tool to help parish and communities to do choices in the liturgy, but in the context of Romcal, official missal (including Lectionaries) must stay the reference. And I'm not speaking of the mainstream missal published every year, but the perpetual missal, used by the priest.

I took a second and closer look to the official Latin missal and the official missal and lectionaries for all French-speaking countries. On all these books, days before epiphany are called with their date (e.g. `Die 2 ianuarii`, `Die 2 ianuarii`... in Latin) in a `ante Epiphaniam` section, and this is normal since these celebrations (especially readings) are defined per date, and are different from the weekdays after Epiphany (which are named `Feria secunda post Epiphaniam`, `Feria tertia post Epiphaniam`...). Only the mass prayers are the same before/after epiphany.

FYI, in the Latin missal, celebrations are organized and titled per date in the "Lectiones ante Epiphaniam" section, and per weekday in another "Lectiones post Epiphaniam" section. And prayers (which are the same before and after epiphany) are gathered into an appendix of weekday prayers, suitable for all the season of Christmas time.

In the French missal, it's the same as the Latin missal, except that prayers have been duplicated and distributed in the days before and after Epiphany (for practical reasons I think, so the priest doesn't have to turn multiple time the pages of the missal). This is the same in the lectionary, except that it contains only the readings (without the prayers).

To keep the spirit of theses special liturgical days, I suggest to stay close and follow the exemple of the missal (both Latin and French), and name the days such as: (`English`, `Latin`)
- before epiphany: `January 2 before Epiphany`, `Die 2 ianuarii ante Epiphaniam`...
- after epiphany: `Monday after Epiphany`, `Feria secunda post Epiphaniam`...

**Comment by tukusejssirs** [comment]
Okay, I agree with you. Thanks for taking a closer look into those books; I have no access to them (except for the Slovak translation of _editio typica altera_ at the parish church; no Latin Missal we use). So, once again: thank you very much! :+1:

As a sidenote: If we try to name the celebrations according to the _editio typica tertia_ of the Roman Missal (if available in a particular language; now I talk esp of the Latin, French and English version of it), it might a good idea to proof all celebrations, esp those from the GRC incl weekdays, names of Sundays, etc.

**Comment by tukusejssirs** [comment]
And this issue is related to #210, as I can’t see a reason why _some_ of the celebrations from the GRC are defined in a different file than `general.ts`.

**Comment by tukusejssirs** [comment]
During the time I copied Czech locale strings from `calendarium-romanum`, I noticed that they [don’t use `{{day}} before Epiphany`](https://github.com/igneus/calendarium-romanum/blob/master/config/locales/en.yml#L2) at all (at [@igneus](https://github.com/igneus) is a theologian and a coder and has bigger team of contributor and he consults everything with liturgists). I think that the the liturgical books just conveniantly gather the days before Epiphany as `days before Epiphany`, however, it is never used in the celebration names (which are, as you have found, in form or `Dies 3 ianuarii`). In romcal, we don’t group the days in the same way as the books (we just use the celebration names), therefore IMHO we should remove `{{day}} before Epiphany` altogether.

Actually, I suggest that we should do the following changes (apart from moving the temporale sections in a `temporale` section; see #229):

<details>
  <summary>Current version with changes in comments</summary>

```js
christmastide: {
  season: 'Christmas Season',
  day: '{{day}} of Christmastide',
  octave: '{{count}} Day within the Octave of the Nativity of the Lord',
  sunday: '{{count}} Sunday of Christmas',
},
epiphany: {
  // `epiphany.season` should be removed
  // When needed, `celebrations.epiphany` should be used instead
  season: 'Epiphany',
  // `epiphany.before` should be removed
  before: '{{day}} before Epiphany',
  // `epiphany.after` should be moved into `christmastide`
  //   and renamed to `after_epiphany`
  after: '{{day}} after Epiphany',
},
lent: {
  season: 'Lenten Season',
  weekday: '{{day}} of the {{week}} week of Lent',
  sunday: '{{week}} Sunday of Lent',
  day_after_ash_wed: '{{day}} after Ash Wednesday',
},
holy_week: {
  // `holy_week.season` should be removed
  // When needed, `ranks.HOLY_WEEK` should be used instead
  season: 'Holy Week',
  // `holy_week.weekday` should be moved into `lent` and renamed to `holy_week_weekday`
  weekday: '{{day}} of Holy Week',
},
paschal_triduum: {
  // `paschal_triduum.season` should be removed
  // When needed, `ranks.TRIDUUM` should be used instead
  season: 'Paschal Triduum',
},
```
</details>

<details>
  <summary>Proposed version with changes implemented</summary>

```js
christmastide: {
  season: 'Christmas Season',
  day: '{{day}} of Christmastide',
  octave: '{{count}} Day within the Octave of the Nativity of the Lord',
  sunday: '{{count}} Sunday of Christmas',
  after_epiphany: '{{day}} after Epiphany',
},
lent: {
  season: 'Lenten Season',
  weekday: '{{day}} of the {{week}} week of Lent',
  sunday: '{{week}} Sunday of Lent',
  day_after_ash_wed: '{{day}} after Ash Wednesday',
  holy_week_weekday: '{{day}} of Holy Week',
},
```
</details>

**Comment by emagnier** [comment]
As the weekdays of Advent between 17-24 December, days before Epiphany can be named with their dates only (January 5) without referring to the actual period 👍 (unlike other weekdays from the temporal cycle). But as you know, romcal is an API and not a final GUI, and it's still important to identify well days within this period, for any app plugged on romcal which need this information (liturgy of the hours, monastic liturgies...).

Also the localisation file you are pointing out is referring to liturgical periods, which include liturgical seasons but not only. According to how the romcal engine is working, we should keep it as it.

> at @ igneus is a theologian and a coder and has bigger team of contributor and he consults everything with liturgists

It seems to be an argument 😄 , but this isn't sufficient for me, even if I recognise the great quality of his projects and contributions (but no project is perfect, starting with romcal). And if you only knew all the liturgists that don't agree together... :roll_eyes: I'm myself a student in theology, specialized in liturgy, and have luckily access to great resources in my university (persons and books). I also have a long practical experience in this field, especially in the monastic world 😉

**Comment by emagnier** [comment]
To summarise the whole discussion:

1. The `christmastideIncludesTheSeasonOfEpiphany` option will be removed (there is no season of Epiphany).

2. A refactor and improvement has already been done about seasons and periods metadata. In addition to the season (`CHRISTMASTIDE`), the liturgical period property will also specify if a liturgical day is within:
    - `CHRISTMAS_OCTAVE`
    - `DAYS_BEFORE_EPIPHANY`
    - `DAYS_AFTER_EPIPHANY`

    This will also address the need for some third apps using romcal, to get more liturgical contextual information.

3. The naming of weekdays in the period of Epiphany will follow these examples:
    - Before epiphany: `January 2`, `Die 2 ianuarii`...
    - After epiphany: `Monday after Epiphany`, `Feria secunda post Epiphaniam`...

Anything else?

**Comment by tukusejssirs** [comment]
> Also the localisation file you are pointing out is referring to liturgical periods, which include liturgical seasons but not only. According to how the romcal engine is working, we should keep it as it.

Even if we don’t remove `epiphany.before`, can’t we just merge `epiphany` into `christmastide`? Like `christmastide.before_epiphany` and `christmastide.after_epiphany`. This is just a change of l10n key location and I think it is more reasonable to have only those seasons in the l10n files which actully exist. Any additional (sub-) seasons should be IMHO located within the official season objects. I still think that all season objects in the l10n files should be moved to `temporal` object (see #229), while keeping the non-seasonal objects (e.g. `ranks`, `liturgical_colors`) would be kept were they are.

> > at @ igneus is a theologian and a coder and has bigger team of contributor and he consults everything with liturgists
>
> It seems to be an argument :smile: , but this isn't sufficient for me, even if I recognise the great quality of his projects and contributions (but no project is perfect, starting with romcal). And if you only knew all the liturgists that don't agree together... :roll_eyes: I'm myself a student in theology, specialized in liturgy, and have luckily access to great resources in my university (persons and books). I also have a long practical experience in this field, especially in the monastic world :wink:

Note that I didn’t want to say that your knowledge is insufficiant or bad. I didn’t (or I forgot) that you study theology.

---

> To summarise the whole discussion:
>
> […]
>
> Anything else?

Well, I think we need to create l10n keys for (change the key name if you find a better name):
- `advent.calendar_days`;
- `christmastide.calendar_days`.

Why? At least because of Latin. The word `Dies` is IMHO not part of the date _per se_, therefore I didn’t add it to the `dayjs` output (I think that in general it can be part of the date format, but does not need to). Therefore the value should be `Dies {{dom_month}}` for Latin; other locales should have it set to `{{dom_month}}`.

Also note that the `{{dom_month}}` in all locales except for Latin should start with capital first letter. `dayjs` output the month name in lowercase unless the month name is always written with capital first letter in a particular locale (like in English). Note that this is important when `{{dom_month}}` starts with month name.

**Comment by tukusejssirs** [comment]
While preparing another proposal to the marthrology table (see #208), it came to my mind that we could have (unofficial) subseasons of these: `epiphany`, `ordinary_time_i`, `ordinary_time_ii`, `paschal_triduum`, `holy_week`. Those could be subobjects to the main seasons.

**Comment by emagnier** [comment]
Addressed on https://github.com/romcal/romcal/pull/250.
